drush/civicrm.drush.inc

```diff
@@ -1605,7 +1607,12 @@ function drush_civicrm_api() {
         unset($params['version']);
         $result = civicrm_api4($entity, $action, $params);
       }
-      catch (API_Exception $e) {
+      catch (DBQueryException $e) {
+        drush_set_error('CIVICRM api error', $e->getDBErrorMessage() . ' ' . $e->getUserMessage()
+          . (in_array('debug=1', $args, TRUE) ? "\n" . $e->getSQL() : '')
```

Comment:
```suggestion
          . (in_array('debug=1', $args, TRUE) ? "\n" . $e->getSQL() : '')
          . (!empty($params['debug']) ? ("\n" . $e->getSQL()) : '')
```
$args has already been parsed and put into params, and depending on how you specify the args it might not look like `debug=1`. In fact I couldn't even figure out how to use this for api4 without using the json method where it's not in args.

Also for me getSQL() returned blank, but that's not a blocker maybe it depends what the error situation is. For me I forced a db syntax error.

I also get a secondary error `Undefined variable $result`, but it's not because of the PR. A followup could wrap the output block below to only run if `$result` actually got set.

Reply:
@demeritcowboy then I think maybe checking debug is too complex & just add the extra output?